tidb-binlog/pump_client/pump.go

```diff
@@ -105,7 +159,7 @@ func (p *PumpStatus) createGrpcClient() error {
 		clientConn, err = grpc.Dial(p.Addr, dialerOpt, grpc.WithInsecure())
 	}
 	if err != nil {
-		atomic.AddInt64(&p.ErrNum, 1)
+		p.state.updateFailedTS(int64(time.Now().Second()))
```

Comment:
// Second returns the second offset within the minute specified by t, in the range [0, 59]. 
look not the p.state want really